core/services/fluxmonitor/flux_monitor.go

```diff
@@ -590,9 +590,15 @@ func (p *PollingDeviationChecker) respondToAnswerUpdatedLog(log *contracts.LogAn
 //
 // Only invoked by the CSP consumer on the single goroutine for thread safety.
 func (p *PollingDeviationChecker) respondToNewRoundLog(log *contracts.LogNewRound) {
-	// The idleThreshold resets when a new round starts
+	// The idleTimer resets when a new round starts
 	if !p.idleThreshold.IsInstant() {
-		p.idleTicker = time.After(p.idleThreshold.Duration())
+		p.idleTimer = MakeNewRoundIdleTimer(log, p.idleThreshold, realClock{})
+		select {
+		case <-p.idleTimer:
+			// TODO: How to handle immediately expired timer here?
```

Comment:
Of course we can get rid of it, I left it here to demonstrate that we may want to handle it specially here.

Reply:
To me it seems pointless to continue with any of the other logic in this function if the round is already expired... but I may well be missing something.

Any ideas? @se3000 @spooktheducks @kaleofduty @coventry ?

Reply:
To be clear, this has nothing to do with the round. Just with the idle timer?
If so, I'd argue the node should run the regular submission logic.

@spooktheducks and @RyanRHall should definitely also weigh in on this.

Reply:
@samsondav an example scenario where I think this could happen is in the event of a reorg. We're currently on block A, but we reorg back to block B, and receive NewRound logs that are already expired because enough time has passed before the reorg.

I think in this scenario the behavior of the node should be to both submit an answer to the expired round _and_ to start a new round. I would second @kaleofduty's suggestion that we don't read from the timeout channel here and instead just let consume handle this like normal. It feels like extra complexity to me to try and handle the `idleTimeout` in two places.